src/aws_encryption_sdk/materials_managers/default.py

```diff
@@ -143,3 +207,55 @@ def decrypt_materials(self, request):
         )
 
         return DecryptionMaterials(data_key=data_key, verification_key=verification_key)
+
+    def _decrypt_materials_using_keyring(self, request):
+        """Obtains a plaintext data key from one or more encrypted data keys
+        using underlying keyring.
+
+        :param request: decrypt materials request
+        :type request: aws_encryption_sdk.materials_managers.DecryptionMaterialsRequest
+        :returns: decryption materials
+        :rtype: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        :raises InvalidCryptographicMaterialsError: if keyring cannot complete decryption materials
+        :raises InvalidCryptographicMaterialsError:
+           if decryption materials received from keyring do not match request
+        """
+        verification_key = self._load_verification_key_from_encryption_context(
```

Comment:
Does this throw if the suite wants a key, but the EC does not have one?

Reply:
Yes.

https://github.com/aws/aws-encryption-sdk-python/blob/952bce44db630754de6a0dad451f72665f83b1c7/src/aws_encryption_sdk/materials_managers/default.py#L116-L117